### DISCUSSION TITLE
Poetry build dependencies managment

### DISCUSSION THREAD

**Discussion by lordisma** [python-poetry/poetry#8715] [Q&A] [answered]
Hi, 

I am working in a project that involved Cython and numpy. When trying to compile the `pyx` files we hit into a problem as the poetry build environment doesn't seems to have `numpy` installed in there. We double checked that `numpy` was added into the poetry dependencies and tried to install it into the global environment. Neither of that approaches solved the issue.  

I made up to create a minimal example creating a `build.py` file and importing `numpy` in there. My question is, how can I add dependencies inside the building environment?

Following snippet of code is an example of the kind of error I am seeing
```
 $ test poetry install
Installing dependencies from lock file

Package operations: 1 install, 0 updates, 0 removals

  • Installing numpy (1.26.2)

Preparing build environment with build-system requirements poetry-core
Traceback (most recent call last):
  File ".../Documents/Projects/test/build.py", line 1, in <module>
    import numpy as np
ModuleNotFoundError: No module named 'numpy'
```

**Comment by finswimmer** [comment, marked as answer]
Hey,

you have to add your build requirements to `requires` in the `[build-system]` section of your `pyproject.toml`. See https://python-poetry.org/docs/pyproject/#poetry-and-pep-517

fin swimmer

**Reply by lordisma** [reply]
Thank you @finswimmer that solved the issue 😄
